javascripts/discourse/components/user-color-palette-selector.gjs

```diff
@@ -0,0 +1,175 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import AsyncContent from "discourse/components/async-content";
+import icon from "discourse/helpers/d-icon";
+import { reload } from "discourse/helpers/page-reloader";
+import {
+  listColorSchemes,
+  loadColorSchemeStylesheet,
+  updateColorSchemeCookie,
+} from "discourse/lib/color-scheme-picker";
+import { i18n } from "discourse-i18n";
+import DMenu from "float-kit/components/d-menu";
+import UserColorPaletteMenuItem from "./user-color-palette-menu-item";
+
+const HORIZON_PALETTES = [
+  "Horizon",
+  "Marigold",
+  "Violet",
+  "Lily",
+  "Clover",
+  "Royal",
+];
+
+export default class UserColorPaletteSelector extends Component {
+  @service currentUser;
+  @service keyValueStore;
+  @service site;
+  @service session;
+  @service interfaceColor;
+
+  constructor() {
+    super(...arguments);
+
+    // Only need to do this for anon because the current user will have their
+    // color scheme cookie set and other automatic things from core loads their
+    // preference.
+    if (!this.currentUser && this.selectedColorPaletteId) {
+      loadColorSchemeStylesheet(this.selectedColorPaletteId);
+    }
+  }
+
+  get userColorPalettes() {
+    return listColorSchemes(this.site);
+  }
+
+  get selectedColorPaletteId() {
+    if (this.currentUser) {
+      return this.session.userColorSchemeId;
+    }
+
+    return this.keyValueStore.getItem("anon-horizon-color-palette-id");
+  }
+
+  @action
+  onRegisterMenu(api) {
+    this.dMenu = api;
+  }
+
+  @action
+  async buildColorPaletteObject() {
```

Comment:
This function is no longer async because all the data neeed is already serialized. 

This logic can be moved to the getter `userColorPalettes` which also fixes a small bug where the selector can be displayed if there are no palettes that are user selectable (see below)